packages/js/product-editor/src/components/editor/init-blocks.ts

```diff
@@ -14,7 +20,14 @@ import { init as initPricing } from '../pricing-block';
 import { init as initCollapsible } from '../collapsible-block';
 
 export const initBlocks = () => {
-	registerCoreBlocks();
+	const coreBlocks = __experimentalGetCoreBlocks();
+	const blocks = coreBlocks.filter( ( block: BlockInstance ) => {
+		return ! getBlockType( block.name );
+	} );
```

Comment:
Nice, I guess this works to :)
I had talked with @mdperez86 about it, and thought the `initBlocks` should more so be something triggered outside of the editor package (same goes for the rich text editor component ), so controlled by the client instead of the editor component.
But this logic would be handy anyhow and fixes the immediate problem. It may still be nice to move the `initBlocks` and `registerCoreBlocks` calls outside of the respective Editor components, but that could be a future change.

Reply:
Ya, I'm also thinking the same.

Part of me feels like there may be performance gains to not initialize blocks on every single WCA page and only do it when these components are loaded, but those gains may be overstated or even non-existent.

But let's leave it for a follow-up and we can think about the best way to enqueue all these blocks/assets for various use cases in WCA.